group/assignment6/move_cars.py

```diff
@@ -0,0 +1,59 @@
+def generate_moves(start, final):
+    """ ith element of start start is the car currently in the ith space
+    ith element of final start is the car that should be in ith space
+    empty space is 0
+    yields optimal sequence of moves in format (car number, old space, new space) """
+
+    # validation of the input
+    start_set = set(start)
+    final_set = set(final)
+    #print(start_set, final_set)
+    if len(start_set) != len(start) or start_set != final_set or \
+       len(final_set) != len(final) or \
+       start_set != final_set and 0 not in start_set:
+        yield None
+
+    # pos will contain car positions in another form
+    pos = [0] * len(start)
```

Comment:
Same thing here about code structure. Should this be a logical part of the generation?

My personal preference is to have it in the same function, but, well
Competitive programming: if you can read it 5 minutes later, it's good code

I'm probably not the best judge here:)